fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java

```diff
@@ -201,6 +198,11 @@ public List<Integer> getSortKeyIndexes() {
         return indexes;
     }
 
+    // day(dt) -> identity dt
+    public boolean hasPartitionTransformedEvolution() {
+        return getNativeTable().spec().fields().stream().anyMatch(field -> field.transform().isVoid());
```

Comment:
Does this mean it use a expr parition?

Reply:
it means iceberg table has alter `partition pattern` on transformed partitioned column. for explame:  `dt -> day(dt)` or `day(dt) -> dt` or `month(dt) -> day(dt) `.